src/core/server/http/http_config.ts

```diff
@@ -106,6 +108,17 @@ export const config = {
         if (!rawConfig.basePath && rawConfig.rewriteBasePath) {
           return 'cannot use [rewriteBasePath] when [basePath] is not specified';
         }
+
+        if (rawConfig.publicBaseUrl) {
+          const parsedUrl = url.parse(rawConfig.publicBaseUrl);
+          if (parsedUrl.query || parsedUrl.hash || parsedUrl.auth) {
+            return `[publicBaseUrl] may only contain a protocol, host, port, and pathname`;
+          }
+          if (parsedUrl.path !== (rawConfig.basePath ?? '/')) {
+            return `[publicBaseUrl] must contain the [basePath]: ${parsedUrl.path} !== ${rawConfig.basePath}`;
+          }
```

Comment:
Does `url.parse('http://myhost.com:4').path === '/'`? I'm unsure, looking at the tests.

Reply:
Yes, the path is `/` when there is no path in the input URL. I'll add a test to make this more clear / ensure it doesn't break

Reply:
Actually I believe this covers this:
```
    expect(() => httpSchema.validate({ publicBaseUrl: 'http://myhost.com' })).not.toThrow();
    expect(() => httpSchema.validate({ publicBaseUrl: 'http://myhost.com/' })).not.toThrow();
```